### PR TITLE
FSEntry changes

### PR DESCRIPTION
**Pull request by Hwesta**
* Move serialization to FSEntry class
* GROUPID optional
* Add checksums to fileSec
* mdRef specifies attribute order
* Setup rest of logging

Should go in after #6

### CODE REVIEW

File: metsrw/fsentry.py [hunk 1]
```diff
@@ -63,6 +74,13 @@ def __init__(self, path, label=None, use='original', type=u'Item', children=None
         self.children = children
         self.file_uuid = file_uuid
         self.derived_from = derived_from
+        if bool(checksum) != bool(checksumtype):
```
Comment (mistydemeo) [inline metsrw/fsentry.py]: Why `bool` instead of directly checking truthiness?

**Comment by mistydemeo** [comment]
:+1:

File: metsrw/fsentry.py (hunk 1 above)
Comment (Hwesta) [inline metsrw/fsentry.py]: I was trying to check that both checksum & checksumtype were passed, or neither.  Obviously if they're both passed they won't be identical values, so I'm trying to check truthiness by converting to bool.

Comment (mistydemeo) [inline metsrw/fsentry.py]: Oh, I see - makes sense.

**Pull request merged**